### DISCUSSION TITLE
Microphone on Windows

### DISCUSSION THREAD

**Discussion by darioxz** [dotnet/maui#3737] [Q&A]
Hi Everyone

I need to process audio signal of the microphone input in real-time.

On Android, it's straightforward using `Android.Media.AudioRecord` and then reading the buffer in a loop with `ReadAsync(...)`. 

On Windows, I found `Windows.Media.Capture.MediaCapture` which should allow me to write the audio signal to an instance of `InMemoryRandomAccessStream`. So far I haven't found a way to read data from this stream and free the memory afterwards. 

As this seems like a standard problem, what is the usual way to do this on Windows?
Do I need to write a c++ Library for this?

Thanks!

**Comment by nogginbox** [comment]
You shouldn't need C++ for this.

I've not tried, but this question on Stack overflow looks like it might be on the money:
https://stackoverflow.com/questions/12352725/how-to-capture-audio-only-using-windows-mediacapture-api

Hopefully should give you a good starting point at least.

**Reply by darioxz** [reply]
Thanks. I tried this approach to fill a `InMemoryRandomAccessStream` with audio, but then failed to process the data. 

It's fine to record and save the recording as a file, but in my use-case, I need to process the audio in (nearly) real-time, e.g. every 30ms. 

Do you have any suggestions how to achieve this with an `InMemoryRandomAccessStream`?

**Reply by nogginbox** [reply]
I don't. You might want to Stackoverflow this as a UWP question. You'll probably get more eyes on it.
